docs/multi-target-framework.md

```diff
@@ -0,0 +1,150 @@
+# Multi-targeting
+
+## Developer needs / scenarios
+
+* Compile against an earlier version of our bindings.
+    * Example: a customer could be producing a NuGet targeting .NET 7, and
+      want to support the initial release of .NET 7. Said customer must then
+      be able to build against the bindings we shipped at the time.
+    * Ref: https://github.com/dotnet/sdk/issues/30103
+* Consume preview packages for a preview version of Xcode.
+    * Ref: https://github.com/xamarin/xamarin-macios/issues/18343
+
+## Developer experience
+
+### TargetFramework=net7.0-ios16.4
+
+Builds with the bindings we released for iOS 16.4 (Xcode 14.3).
+
+### TargetFramework=net7.0-ios
+
+Builds with the default bindings. This can change in any release (this is
+contrary to what other platforms do - specifically because older OS bindings
+might not work with newer Xcodes, and Apple auto-updates people's Xcodes, so
+it's rather frequent that customers use newer OS bindings. Having a lot of
+people specify the OS version in the target framework is undesirable).
+
+### TargetFramework=net8.0-ios17.0
+
+Builds with the bindings we've released for iOS 17.0 (on .NET 8).
+
+This might be preview bindings, in which case customers must also set the
+following property in their project file to make their intention clear:
+
+```xml
+<PropertyGroup>
+    <EnablePreviewFeatures>true</EnablePreviewFeatures>
+</PropertyGroup>
+```
+
+This mirrors how it's done for the Android SDK.
+
+### TargetFrameworks=net7.0-ios16.0;net8.0-ios-17.0
+
+This would only be valid for a library project.
+
+### Implementation details
+
+We'll release a single workload (per platform), with bindings for every OS
+version we currently support or plan to support. This means our stable
+releases may point to preview packages (but these preview packages have to be
+opted in by doing two things: appending the OS version to their target
+framework + setting EnablePreviewFeatures=true)
+
+We'll rename our packages. Currently we ship these packages:
+
+* Microsoft.iOS.Sdk
+* Microsoft.iOS.Ref
+* Microsoft.iOS.Runtime.ios-arm64
+* Microsoft.iOS.Runtime.iossimulator-arm64
+* Microsoft.iOS.Runtime.iossimulator-x64
+
+and now we'll add the target framework supported by these packages:
+
+* Microsoft.iOS.Sdk.net7.0_16.4
+* Microsoft.iOS.Ref.net7.0_16.4
+* Microsoft.iOS.Runtime.ios-arm64.net7.0_16.4
+* Microsoft.iOS.Runtime.iossimulator-arm64.net7.0_16.4
+* Microsoft.iOS.Runtime.iossimulator-x64.net7.0_16.4
```

Comment:
For android these were:
```
Microsoft.Android.Ref.33
Microsoft.Android.Ref.34
Microsoft.Android.Runtime.33.android-arm64
Microsoft.Android.Runtime.34.android-arm64
```
And then the same SDK pack works for all of them.

The name probably doesn't matter much, but just wanted to share what we have.

Reply:
I think you don't necessarily need to put both the .NET version and the platform version in the package ID.  You could just put one of them in the package ID and use the package version to represent the other one.  It's probably also fine to put both of them in the package ID if that makes things simpler.

Reply:
For Android, we actually need the flexibility 33/34 on the same major .NET version. Android 35 might come out before .NET 9 is released, for example.